src/app/modal/modal-close/modal-close.directive.ts

```diff
@@ -7,11 +7,11 @@ import { MzModalComponent } from '../modal.component';
 })
 export class MzModalCloseDirective {
 
-  @HostListener('click') onclick() {
-    this.modalComponent.close();
-  }
-
   constructor(
     private modalComponent: MzModalComponent,
   ) { }
+
+  @HostListener('click') onclick() {
+    this.modalComponent.close();
+  }
```

Comment:
not sure i like the fact of putting @Hostlistener after the constructor ... I always liked to have those on top, getting a better overview with @input and @output of all bindings! Is it really a best practices to put them after the constructor?

Reply:
My bad...
In the latest Angular-cli (v1.7.2) the `member-ordering` order value is:

```json
"static-field",
"instance-field",
"static-method",
"instance-method"
```

That can translate to (as the schema here represent https://palantir.github.io/tslint/rules/member-ordering/ ; also after it's more a choice of what order you prefer `public`, `protected` and `private`):

```json
"public-static-field",
"protected-static-field",
"private-static-field",
"public-instance-field",
"protected-instance-field",
"private-instance-field",
"public-static-method",
"protected-static-method",
"private-static-method",
"public-instance-method",
"protected-instance-method",
"private-instance-method"
```

Not what I used yesterday (that consider constructor position and reverse some sections):

```json
"public-static-field",
"protected-static-field",
"private-static-field",
"public-static-method",
"protected-static-method",
"private-static-method",
"public-instance-field",
"protected-instance-field",
"private-instance-field",
"public-constructor",
"protected-constructor",
"private-constructor",
"public-instance-method",
"protected-instance-method",
"private-instance-method"
```

Overall, fixing the rule above allow to place `@HostListener` before `constructor`.